packages/trader/src/AppV2/Hooks/useContractsForCompany.ts

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { useStore } from '@deriv/stores';
+import { cloneObject, getContractCategoriesConfig, getContractTypesConfig, WS } from '@deriv/shared';
+
+type TContractType = {
+    text?: string;
+    value: string;
+};
+
+export type TContractTypesList = {
+    [key: string]: {
+        name: string;
+        categories: DeepRequired<TContractType[]>;
+    };
+};
+
+const useContractsForCompany = () => {
```

Comment:
will this hook be used in more than one place? if not we'll trigger contractsforcompany call unnecessarily every time  we want to use `contract_types_list` from hook

Reply:
I think under some conditions we do need to retrigger the hook in the future, but we don't need to call it every time we want `contract_types_list `, the list is saved into the store and we can get it in other places from the store `contract_types_list_v2`. Unless there's some obvious dependency which I'm missing, I don't think it calls contracts_for_company multiple times